store/postgres/src/subgraph_store.rs

```diff
@@ -1056,61 +1060,118 @@ impl WritableStore {
             site,
         })
     }
+
+    fn retry<T, F>(&self, op: &str, f: F) -> Result<T, StoreError>
```

Comment:
Is there a possibility where all shards are down and we keep multiple loops like this one running trying to execute queries, until the database goes back up and we overload it with too many queries?

Also I imagine the number of connections requested at the same time might be huge, but I'm not completely sure, maybe I understood something wrong conceptually.

**Edit**: I've seen you removed `get_with_timeout_warning`, so connection might not be an issue here.

Reply:
Requesting too many connections isn't an issue since most of them will just block on the pool until one becomes available.

Not sure what happens if the shard has been down for a long time - there is a chance of a thundering herd, we'll have to see. Since the backoff is 10s after a while, I'd hope that the retries are splayed reasonably.